### PR TITLE
Bump hikari to 2.0.0 (stable)

### PR DESCRIPTION
**Pull request by lukasthaler**
This PR bumps the required hikari version to the stable 2.0.0 release, matches its supported Python version and makes the required changes to keep the pipelines green

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -55,18 +55,18 @@ def parse_requirements_file(path):
         ':sys_platform=="win32"': ["colorama"],
         "docs": parse_requirements_file("doc_requirements.txt"),
     },
-    python_requires=">=3.8.0,<3.13",
+    python_requires=">=3.9.0,<3.14",
```
Comment (hypergonial) [inline setup.py:58]: 3.13 isn't out yet, please revert the upper bound change

File: setup.py [hunk 2]
```diff
@@ -55,18 +55,18 @@ def parse_requirements_file(path):
...
         "Framework :: AsyncIO",
         "Intended Audience :: Developers",
         "Natural Language :: English",
         "Topic :: Software Development :: Libraries :: Python Modules",
-        "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
         "Programming Language :: Python :: 3.10",
         "Programming Language :: Python :: 3.11",
         "Programming Language :: Python :: 3.12",
+        "Programming Language :: Python :: 3.13",
```
Comment (hypergonial) [inline setup.py:69]: and remove this

File: toolbox/commands.py [hunk 3]
```diff
@@ -26,33 +26,27 @@ def _list_to_command_choices(
 
 
 @t.overload
-def as_command_choices(choices: t.Sequence[ChoiceTypes]) -> t.Sequence[hikari.CommandChoice]:
```
Comment (hypergonial) [inline toolbox/commands.py:29]: Not sure why the formatting changed in this file, out of scope for this PR

File: toolbox/strings.py [hunk 4]
```diff
@@ -1,7 +1,8 @@
 import datetime
 import re
 import typing as t
-from enum import Enum, IntFlag
```
Comment (hypergonial) [inline toolbox/strings.py:4]: Same here

**Review by hypergonial — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (lukasthaler) [reply setup.py:58]: That'd clash with hikari's range again, which was the reason for this PR in the first place. At this point, I'd rather wait until 3.13 releases next week tbh

File: toolbox/commands.py (hunk 3 above)
Comment (lukasthaler) [reply toolbox/commands.py:29]: Will do when I get home

File: setup.py (hunk 1 above)
Comment (lukasthaler) [reply setup.py:58]: With Python 3.13 now released, I consider this resolved

File: setup.py (hunk 2 above)
Comment (lukasthaler) [reply setup.py:69]: Same here

**Review by hypergonial — APPROVED** [review]
Thanks!

**Pull request merged**